swarm/shed/db.go

```diff
@@ -126,5 +151,182 @@ func (db *DB) WriteBatch(batch *leveldb.Batch) (err error) {
 
 // Close closes LevelDB database.
 func (db *DB) Close() (err error) {
+	close(db.quitChan)
 	return db.ldb.Close()
 }
+
+// Meter configures the database metrics collectors
+func (db *DB) Meter(prefix string) {
+	// Initialize all the metrics collector at the requested prefix
+	db.compTimeMeter = metrics.NewRegisteredMeter(prefix+"compact/time", nil)
```

Comment:
Shouldn't *Meter fields assignments be under a lock?

Reply:
I don't think so. Why would they be under a lock? `Meter` is called once, when you initialise the DB, why would you initialise the same DB under different threads?

Reply:
As Meter is exported, it can be called any number of times, this is why I suggested not to export it, in the commend bellow.

Reply:
You're right, `Meter` should not be exported. It should not be called a number of times, but only once.